### PR TITLE
…

### PR DESCRIPTION
**Pull request by jgrant22**
Data bleed caused by missing segment in the message yaml. 
Same thing for Encounter.diagnosis.use

This PR includes fixes for the two threads in slack from the test team.
Both fixes were simple I completely over thought the first issue which is why I moved it here to start clean.

**Problem 1**: One of our testers discovered a scenario where an Identifier with no system still had a system.  If PV1.4 is available then we use it for the system in this case, PV1.4 was not available but we still got a system
```
what we got:
value: 1001918
system: urn:id:1001918
expected:
value: 1001918
```

**Problem 2**:  One of our testers discovered a scenario we were missing a few mappings and that Encounter.diagnosis.use was not cycling through each DG1 segment

### CODE REVIEW

File: src/main/resources/hl7/codesystem/v2ToFhirMapping.yml [hunk 1]
```diff
@@ -352,6 +352,8 @@ V3ReligiousAffiliation:
 
 DiagnosisRole:
    A: AD
+   F: DD # F stands for final DD stands for discharge Diagnosis which is the final diagnosis before being discharged
+   W: CC # When the patient presents with a chief complaint, such as cough, the provider considers the most common diseases that present with cough, forming a working differential diagnosis list
```
Comment (jgrant22) [inline src/main/resources/hl7/codesystem/v2ToFhirMapping.yml:357]: I left comments here to explain my logic. definitely open for discussion or even removal

File: src/main/resources/hl7/message/ADT_A03.yml [hunk 2]
```diff
@@ -30,18 +30,20 @@ resources:
...
       isReferenced: true
       additionalSegments:
-             - PV2
-             - EVN
-             - MSH
-             - OBX
+        - PV2
+        - EVN
+        - MSH
+        - DG1
```
Comment (cragun47) [inline src/main/resources/hl7/message/ADT_A03.yml:36]: ADT_A04, ADT_A28, ADT_A31 look like they might fail the same way?
@LisaWellman

**Review by cragun47 — APPROVED** [review]
1). Please adjust PR description so that external visitors have access to all links and desicriptions.

2).  Pleas check ADT_A04, ADT_A28, ADT_A31.  They look like the _might_ have the same issue from cut and paste.   Correct as needed, update tests.

File: src/main/resources/hl7/codesystem/v2ToFhirMapping.yml [hunk 3]
```diff
@@ -352,6 +352,8 @@ V3ReligiousAffiliation:
 
 DiagnosisRole:
```
Comment (LisaWellman) [inline src/main/resources/hl7/codesystem/v2ToFhirMapping.yml:353]: Can you add a comment this maps from DG1.6 to Encounter.diagnosis.use

File: src/main/resources/hl7/codesystem/v2ToFhirMapping.yml (hunk 1 above)
Comment (LisaWellman) [reply src/main/resources/hl7/codesystem/v2ToFhirMapping.yml:357]: I would not map F, W, but leave them intact in the JSON (the JSON should contain the original code).  This is a general design change we have been discussing, it should likely be deferred until we tackle it holistically.  If we do defer, this issue will be an outstanding issue (that 'F' is dropped).

File: src/main/resources/hl7/codesystem/v2ToFhirMapping.yml (hunk 3 above)
Comment (jgrant22) [reply src/main/resources/hl7/codesystem/v2ToFhirMapping.yml]: added!

File: src/main/resources/hl7/message/ADT_A03.yml (hunk 2 above)
Comment (jgrant22) [reply src/main/resources/hl7/message/ADT_A03.yml:36]: updated the messages!

File: src/main/resources/hl7/codesystem/v2ToFhirMapping.yml (hunk 1 above)
Comment (LisaWellman) [reply src/main/resources/hl7/codesystem/v2ToFhirMapping.yml:355]: This is what is in the mapping document:
```
F,W have to FHIR equivalent, use the source system http://terminology.hl7.org/CodeSystem/v2-0052
and display (Final, Working)
```

**Pull request merged**